gateway/webrtc/client.js

```diff
@@ -60,6 +61,37 @@ $(document).ready(function() {
     connectToSignallingJMS();
 });
 
+function showInCall() {
+    callBtn.style.display='none';
+    callToUsernameInput.style.display='none';
+    hangUpBtn.style.display='inline';
+    overlay.style.visibility='visible';
+    overlay.innerText='Connected to '+connectedUser;
+}
+
+function showOffCall() {
+    callBtn.style.display='inline';
+    callToUsernameInput.style.display='inline';
+    hangUpBtn.style.display='none';
+    overlay.style.visibility='hidden';
+}
+
+function reconstructSDP (sdp) {
+    var result = [];
+
+    var lines = sdp.split("\n");
+    for (let x of lines) {
+        if (x.indexOf("candidate") > 0 && x.indexOf("typ host") > 0) {
```

Comment:
Candidates are sent not only via "candidate" messages, using Trickle ICE, but also via offer/answer exchanges. A SDP for offer answer looks like this. 


> v=0
> o=alice 20518 0 IN IP4 0.0.0.0
> s=
> t=0 0
> a=ice-ufrag:074c6550
> a=ice-pwd:a28a397a4c3f31747d1ee3474af08a068
> a=fingerprint:sha-1 99:41:49:83:4a:97:0e:1f:ef:6d:f7:c9:c7:70:9d:1f:66:79:a8:07
> a=rtcp-rsize
> m=audio 54609 RTP/SAVPF 0 109 98
> c= IN IP4 24.23.204.141
> a=rtpmap:0 PCMU/8000
> a=rtpmap:109 opus/48000
> a=ptime:20
> a=rtpmap:98 iLBC/8000 a=fmtp:98 mode=20
> a=sendrecv
> a=rtcp-mux
> b=AS:256
> b=RS:0
> b=RR:0
> a=candidate:0 1 UDP 2113667327 192.168.1.4 54609 typ host
> a=candidate:1 1 UDP 694302207 24.23.204.141 54609 typ srflx raddr 192.168.1.4 rport 54609
> a=candidate:0 2 UDP 2113667326 192.168.1.4 64678 typ host
> a=candidate:1 2 UDP 1694302206 24.23.204.141 64678 typ srflx raddr 192.168.1.4 rport 64678
> a=rtcp-fb:109 nack
> m=video 62537 RTP/SAVPF 99 120
> c= IN IP4 24.23.204.141
> a=rtpmap:99 H264/90000
> a=fmtp:99 profile-level-id=4d0028;packetization-mode=1
> a=rtpmap:120 VP8/90000
> a=sendrecv
> a=rtcp-mux
> a=candidate:0 1 UDP 2113667327 192.168.1.4 62537 typ host
> a=candidate:1 1 UDP 1694302207 24.23.204.141 62537 typ srflx raddr 192.168.1.4 rport 62537
> a=candidate:0 2 2113667326 192.168.1.4 54721 typ host
> a=candidate:1 2 UDP 1694302206 24.23.204.141 54721 typ srflx raddr 192.168.1.4 rport 54721
> a=rtcp-fb:99 nack pli
> a=rtcp-fb:99 ccm fir
> a=rtcp-fb:120 nack pli
> a=rtcp-fb:120 ccm fir
> m=application 56966 DTLS/SCTP 5000
> c= IN IP4 24.23.204.141
> a=sctpmap:5000 webrtc-Datachannel 1
> a=webrtc-Datachannel:5000 stream=1;label="channel 1";subprotocol="chat";
> a=sendrecv
> a=rtcp-mux
> **a=candidate:0 1 UDP 2113667327 192.168.1.7 56966 typ host**
> **a=candidate:1 1 UDP 1694302207 24.23.204.141 56966 typ srflx raddr 192.168.1.7 rport 56966**
> **a=candidate:0 2 UDP 2113667326 192.168.1.7 51641 typ host**
> **a=candidate:1 2 UDP 1694302206 24.23.204.141 51641 typ srflx raddr 192.168.1.7 rport 51641**
> a=rtcp-fb:5000 nack

The code simply removes any "host" candidate, in the offer/answer SDP. Otherwise we would expose the local IP via signalling, although it is not exposed via turn/stun protocol (because of the turn proxy)